packages/stark-ui/src/modules/input-mask-directives/directives/text-mask-base.directive.ts

```diff
@@ -0,0 +1,139 @@
+import { AnyMaskedOptions } from "imask";
+import { StarkNumberMaskConfig } from "./number-mask-config.intf";
+import { StarkTextMaskBaseConfig } from "./text-mask-config.intf";
+import { IMaskDirective, IMaskFactory } from "angular-imask";
+import {
+	AfterViewInit,
+	ElementRef,
+	Inject,
+	Injectable,
+	OnChanges,
+	OnDestroy,
+	Optional,
+	PLATFORM_ID,
+	Renderer2,
+	SimpleChange,
+	SimpleChanges
+} from "@angular/core";
+import { COMPOSITION_BUFFER_MODE } from "@angular/forms";
+import { StarkTimestampMaskConfig } from "./timestamp-mask-config.intf";
+
+export type MaskConfigType = StarkTextMaskBaseConfig | StarkNumberMaskConfig | StarkTimestampMaskConfig | string | boolean;
+
+@Injectable()
+export abstract class TextMaskBaseDirective<
```

Comment:
Could you rename this class to `AbstractStarkTextMaskBaseDirective` ?
The file should be renamed to `abstract-text-mask-base-directive.ts`